### PR TITLE
Issue/6714 media picker redesign p1

### LINKED ISSUE #6714: Media Picker Redesign, Part 1
* Single tap no longer inserts the tapped item, instead it selects it
* Long press to show media detail
* Selection circle on every item (to indicate multi-select is possible)

### PR DESCRIPTION
**Pull request by nbradbury**
Resolves #6714 - Updates the editor's media picker as follows:

* Single tap no longer inserts the tapped item, instead it selects it
* Long press to show media detail
* Selection circle on every item (to indicate multi-select is possible)
* Removed the preview icon
* Changes the ActionMode style to make it more obvious that you insert from the top toolbar

GIF demo below, higher quality video [here](https://cloudup.com/c-5r3iuMwpM).

### CODE REVIEW

File: WordPress/src/main/res/layout/photo_picker_thumbnail.xml [hunk 1]
```diff
@@ -17,39 +17,32 @@
             android:layout_height="wrap_content"
             android:background="@drawable/photo_picker_item_background"
             android:scaleType="centerCrop"
-            tools:src="@drawable/gravatar_placeholder" />
+            tools:layout_height="match_parent"
+            tools:layout_width="match_parent"
+            tools:src="@drawable/gravatar_placeholder"/>
 
         <org.wordpress.android.widgets.WPTextView
```
Comment (daniloercoli) [inline WordPress/src/main/res/layout/photo_picker_thumbnail.xml]: Should we use a TextView here, can't be a simple TextView used instead?

**Comment by daniloercoli** [comment]
I'm happy to see the SmartToast gone 💯
A little worried about the long press to show media details...but luckily we've the `MEDIA_PICKER_PREVIEW_OPENED` event bumped, and can compare trends in month or so.

**Comment by nbradbury** [comment]
> Should we use a TextView here

Yes - changed in 9eea6ba

**Comment by nbradbury** [comment]
> A little worried about the long press to show media details...but luckily we've the MEDIA_PICKER_PREVIEW_OPENED event bumped, and can compare trends in month or so.

I'm worried about that, too, but I also didn't like having that ugly preview icon on every image.

**Comment by iamthomasbishop** [comment]
Awesome work, @nbradbury! I've got a couple bits of feedback:

* Status bar: any chance we can get the status bar to be a slightly darker shade than the top toolbar (currently it's slightly lighter)?
* Can we ditch the white border on the selected (count) circle, add a shadow, etc (details here, under Selection Dots: https://goo.gl/8wk8xG). Might also need to be slightly larger to accommodate for two-digit counts.
* Can we rotate the `(+)` icon 45° clockwise when the picker is open, and back (to default position) -45° counter-clockwise when it's closed?
* Probably meant for a diff pr, but we should probably update video/doc/audio cells to match the iOS redesign as well

Great stuff, as always!

**Comment by nbradbury** [comment]
> Status bar: any chance we can get the status bar to be a slightly darker shade than the top toolbar (currently it's slightly lighter)?

How about this?

status

> Can we ditch the white border on the selected (count) circle, add a shadow, etc 

I ditched the border and added a shadow, but I'm not able to add a shadow like the one in that doc - in Android they're more subtle. Here's how it looks:

shadow

> Can we rotate the (+) icon 45° clockwise 

I meant to mention that! That will need to be in a separate PR on the editor project since that toolbar is part of the editor.

**Comment by iamthomasbishop** [comment]
@nbradbury The status/toolbar looks good, as well as the selection dots! One minor thing, can we make it so the shadow on the selection dots isn't cut off by the cell?

Sounds good re: the rotation on the `(+)` icon!

**Comment by nbradbury** [comment]
> One minor thing, can we make it so the shadow on the selection dots isn't cut off by the cell?

@iamthomasbishop Didn't notice that! I decreased the size of the shadow - is this better?

screenshot_1508183443

**Comment by iamthomasbishop** [comment]
Looks good!

**Comment by daniloercoli** [comment]
:shipit:

**Pull request merged**